sdk/servicebus/Azure.Messaging.ServiceBus/src/Client/ServiceBusClientOptions.cs

```diff
@@ -49,6 +51,23 @@ public ServiceBusRetryOptions RetryOptions
                 _retryOptions = value;
             }
         }
+
+        /// <summary>
+        /// The list of plugins for the client.
+        /// </summary>
+        internal List<ServiceBusPlugin> Plugins { get; set; } = new List<ServiceBusPlugin>();
```

Comment:
Any reason why we wouldn't want this to be public?   It may be nice to be able to set the list explicitly rather than add one-by-one.  Likewise for walking the list to see what's in it.    I'm good either way; just a passing thought.

Reply:
Mostly for evolvability purposes. Also we do validate that the passed in plugin is not null. In the future, we might want to do other validation or change implementation details.